oeps/oep-0052-proc-package-release.rst

```diff
@@ -0,0 +1,71 @@
++---------------+--------------------------------------------------------------+
+| References    |                                                              |
++---------------+--------------------------------------------------------------+
+
+Abstract
+========
+
+This document proposes a standard package release process to be followed across the edX community.
+
+Motivation
+==========
+
+We recently shifted the CI from Travis to Github Actions which led us to move the package release process to Github Actions as well. While making this change, we encountered different approaches used for this task in different packages, leading to confusions. As new releases for packages are published from time to time by the edX community, we need to ensure that a consistent approach is used by edX engineers to make these releases. Here we have documented a standard approach for this purpose.
+
+Python Package Release
+======================
+
+Setting up the Github Action
+____________________________
+
+1. Go to the Actions tab in your package’s repository. Click on the ‘New Workflow’ button.
+2. You’ll get a list of available Github workflow templates. Choose the ‘PyPI Package release Workflow’ by clicking on ‘Set up this workflow’
+3. Now click on the ‘Start Commit’ button at the right side of the screen. A dialog will appear. Write the commit message and description. Choose whether you want to merge it to master or create a new branch and start a PR. Then click on ‘Commit New File’.
+4. Once this workflow is up on master, whenever you create a new Github Release for the repository, the above created Github Action will trigger and create a package release on PyPI.
+
+Triggering a New Release
+________________________
+
+1. Make sure to bump the version number in the setup.py file.
+2. Commit your code and tag the commit with the new version number you just specified.
+3. Create a github release on this tag and mention a summary and description for the release.
+4. Once you publish the Github Release, the release on PyPI will also get triggered.
+
+Github Release Summary and Description
+======================================
+
+* Summary is just a one line description of the major change made in the release.
+* Github release description should contain the following:
+   * New features
+   * Bug fixes
+   * Breaking changes
+   * Improvements
+* It is a good practice to use plain language and add any additional resources if needed to make it more comprehensive
+
+Rejected Alternatives
+=====================
+
+We considered triggering a PyPI release on Tag creation as we used to do in Travis. But we observed that engineers sometimes forget to create a Github Release when they tag a commit so we have some PyPI releases without any Github release in parallel. As the best practice is to create a Github Release whenever a PyPI release is triggered, we selected Github Release as the event triggering the action instead of Tagging.
```

Comment:
I wonder why it's important to have a GitHub release if we already have PyPI packages?  What is looking at the GitHub releases?

Personally, I'd prefer to have a changelog in the repository, and a PyPI package, and be done.  GitHub Releases seem like an extra step that doesn't add anything.

Reply:
As far as I'm aware, the main features of GitHub Releases are:

* It's a standard place where many people go to look for release information
* Files can be associated with them (but we rarely utilize this)
* Release notes can be associated with them (but this can be confusing if there's a separate changelog file with different content)
* The creation of a release sends an event that can trigger a GitHub Action
* Users can subscribe to notifications specifically for new GitHub Releases in a repo

So...kind of a mixed bag, in my opinion.  They're useful to have as an automation hook, and I look at them pretty often because many projects put more care into their GitHub Releases than their changelog (if they even have a separate changelog file).  But for many projects the killer feature is the ability to attach tarballs and binaries, and we don't really need that.  I was leaning towards "useful to have them", but I'm definitely interested in hearing other people's views on whether we should use them or not.

Reply:
It's important to me that the changelog be available in the repo itself.  For example, if I clone a repo, I should be able to examine the changelog locally.  If a GitHub Release is a mirror of the repo file, or vice-versa, that works for me. I definitely don't want people to be told they have to go to github.com to read about what has changed. 

In the Python world, do people look for GitHub Releases? I go to PyPI to see what versions are available for a package.

Reply:
I'd definitely prefer to have the primary changelog be in the code, and just copy that into the GitHub Releases page when making the release.  I can still imagine them getting out of sync if somebody later updates one of those locations (when fixing a typo or broken link, etc.) and forgets to update the other one, though.  We could probably automate keeping them synchronized if it's important enough.

Looking at https://openedx.atlassian.net/wiki/spaces/AC/pages/1001521320/Python+Package+Changelogs , I see 44 packages used in edX code for which the GitHub Releases page was the most accurate source of release notes the last time we looked.  I think I've done the bulk of contributions to that page, and as noted at the top of the doc I generally prefer to use any current changelog in RTD or the repo over the GitHub Releases page.  But some packages just use GitHub Releases as their authoritative changelog.

Reply:
For what it's worth, I almost exclusively look at GitHub releases to understand the change logs of npm packages, and the vast majority seem to use the feature and to host their code on GitHub.  

I do think this conversation is subjective, though, and we should be careful to separate out our own workflows from industry best practices.  Other folks may look at npm itself, hunt for a changelog file, or look for it in documentation first.  I've just found that I seem to have the most luck in finding a changelog the first time when I start by looking at an npm package's GitHub releases.

Today the change logs for our frontend libraries are generated from conventional commit messages and automatically formatted into the description of our GitHub releases.  I don't feel super opinionated about what we end up doing, but I _do_ think it should reflect what the rest of the ecosystem does, and I do think it should be as automated as possible.  For npm packages, my relatively confident opinion is that we're already there.

Reply:
I'm with @nedbat, here.  The source of truth should be CHANGELOG in the repo, and anything else should be considered nice-to-have and synced to _that_.  The simplest defense is that this allows a repository to be moved to other source code management tools with minimal hassle.  (Vendor lock-in sucks.)

As for whether to have GH releases along with PyPI releases, I agree that having both synced-up is ideal, as long as the cost is manageable.